substrate/frame/treasury/src/lib.rs

```diff
@@ -288,6 +288,9 @@ pub mod pallet {
 
 		/// Provider for the block number. Normally this is the `frame_system` pallet.
 		type BlockNumberProvider: BlockNumberProvider;
+
+		#[pallet::constant]
+		type BlockNumberProviderName: Get<&'static str>;
```

Comment:
```suggestion
		#[pallet::constant]
		type BlockNumberProvider: BlockNumberProvider;
```

The idea was that you could do this.

Reply:
Maybe what we should do is the following.

```
trait BlockNumberProvider {
       const IDENTIFIER: &'static str;
       ...
}
```

```
#[pallet::constant(::IDENTIFIER)]
```

We extend `constant` to being able to choose where to fetch the value from.